modules/home-manager/zsh-syntax-highlighting.nix

```
@@ -1,18 +1,39 @@
+# In this file we define both `catppuccin.zsh-syntax-highlighting` and `programs.zsh.syntaxHighlighting.catppuccin` options this is beacuse
+# we have to recreate the lib.mkRenamedOptionModule here due to the use of a submodule being used for `programs.zsh.syntaxHighlighting`
+# which would lead to an error if we used the original `lib.mkRenamedOptionModule`.
+#  see <https://github.com/catppuccin/nix/pull/371#discussion_r1835785145>
 { config, lib, ... }:
 let
   inherit (config.catppuccin) sources;
   inherit (lib) ctp;
-  cfg = config.programs.zsh.syntaxHighlighting.catppuccin;
+  cfg = config.catppuccin.zsh-syntax-highlighting;
   enable = cfg.enable && config.programs.zsh.syntaxHighlighting.enable;
+
+  oldCfg = config.programs.zsh.syntaxHighlighting.catppuccin;
 in
 {
-  options.programs.zsh.syntaxHighlighting.catppuccin = ctp.mkCatppuccinOpt {
-    name = "Zsh Syntax Highlighting";
+  options = {
+    catppuccin.zsh-syntax-highlighting = ctp.mkCatppuccinOpt {
+      name = "Zsh Syntax Highlighting";
+      enableDefault = config.catppuccin.enable || oldCfg.enable;
+    };
+
+    programs.zsh.syntaxHighlighting.catppuccin = ctp.mkCatppuccinOpt {
+      name = "Zsh Syntax Highlighting";
+    };
```

set `enableDefault` to `false` so that the warning doesn't get triggered if `catppuccin.enable` is set to `true`

```suggestion
      enableDefault = false;
    };
```